Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contest/ContestDetailsPage.tsx

```diff
@@ -0,0 +1,263 @@
+import React, { useCallback, useEffect, useMemo } from 'react';
+import isNil from 'lodash/isNil';
+
+import { ContestParticipationType } from '../../common/constants';
+import { IContestDetailsProblemType, IProblemResourceType } from '../../common/types';
+import { ButtonState, LinkButton, LinkButtonType } from '../../components/guidelines/buttons/Button';
+import Heading, { HeadingType } from '../../components/guidelines/headings/Heading';
+import List from '../../components/guidelines/lists/List';
+import SpinningLoader from '../../components/guidelines/spinning-loader/SpinningLoader';
+import ProblemResource from '../../components/problems/problem-resource/ProblemResource';
+import { useRouteUrlParams } from '../../hooks/common/use-route-url-params';
+import { useAppUrls } from '../../hooks/use-app-urls';
+import { useAuth } from '../../hooks/use-auth';
+import { useCurrentContest } from '../../hooks/use-current-contest';
+import { flexCenterObjectStyles } from '../../utils/object-utils';
+import { makePrivate } from '../shared/make-private';
+import { setLayout } from '../shared/set-layout';
+
+import styles from './ContestDetailsPage.module.scss';
+
+const compareByOrderBy = (p1: IContestDetailsProblemType, p2: IContestDetailsProblemType) => p1.orderBy - p2.orderBy;
+
+const ContestDetailsPage = () => {
+    const { state: { params } } = useRouteUrlParams();
+    const {
+        state:
+            {
+                contestDetails,
+                contestDetailsError,
+                contestDetailsIsLoading,
+                isContestDetailsLoadingSuccessful,
+                isUserParticipant,
+            },
+        actions: { getContestDetails },
+    } = useCurrentContest();
+    const {
+        getParticipateInContestUrl,
+        getContestResultsUrl,
+        getAdministrationContestProblemsInternalUrl,
+        getAdministrationContestEditInternalUrl,
+    } = useAppUrls();
+    const { state: { user } } = useAuth();
+
+    const { contestId, participationType } = params;
+
+    const contestIdToNumber = useMemo(
+        () => Number(contestId),
+        [ contestId ],
+    );
+
+    const isOfficial = useMemo(
+        () => participationType === ContestParticipationType.Compete,
+        [ participationType ],
+    );
+
+    useEffect(
+        () => {
+            console.log(params);
+            if (isNil(contestDetails) && isNil(contestDetailsError) && !contestDetailsIsLoading && !isNil(contestId) &&
+            !isContestDetailsLoadingSuccessful) {
+                console.log('TEST');
+                getContestDetails({ id: contestId.toString(), isOfficial });
+            }
+        },
+        [ contestId,
+            getContestDetails,
+            contestDetailsError,
+            contestDetailsIsLoading,
+            contestDetails,
+            isContestDetailsLoadingSuccessful,
+            params,
+            isOfficial,
+        ],
+    );
+
+    const isUserAdmin = useMemo(
```

Comment:
Do you really need this memo? You can use canAccessAdministration instead.

Reply:
It is really not needed, so I removed it.